dynamiq/nodes/writers/pgvector.py

```diff
@@ -0,0 +1,79 @@
+from typing import Any, Literal
+
+from dynamiq.connections import PGVector
+from dynamiq.nodes.node import NodeGroup, VectorStoreNode, ensure_config
+from dynamiq.runnables import RunnableConfig
+from dynamiq.storages.vector.base import BaseWriterVectorStoreParams
+from dynamiq.storages.vector.pgvector.pgvector import PGVectorStore
+from dynamiq.utils.logger import logger
+
+
+class PGVectorDocumentWriter(VectorStoreNode, BaseWriterVectorStoreParams):
+    """
+    Document Writer Node using PGVector Vector Store.
+
+    This class represents a node for writing documents to a PGVector Vector Store.
+
+    Attributes:
+        group (Literal[NodeGroup.WRITERS]): The group the node belongs to.
+        name (str): The name of the node.
+        connection (PGVector | None): The PGVector connection.
+        vector_store (PGVectorStore | None): The PGVector Vector Store instance.
+    """
+
+    group: Literal[NodeGroup.WRITERS] = NodeGroup.WRITERS
+    name: str = "PGVectorDocumentWriter"
+    connection: PGVector | str | None = None
+    vector_store: PGVectorStore | None = None
+
+    def __init__(self, **kwargs):
+        """
+        Initialize the PGVectorDocumentWriter.
+
+        If neither vector_store nor connection is provided in kwargs, a default PGVector connection is created.
+
+        Args:
+            **kwargs: Arbitrary keyword arguments.
+        """
+        if kwargs.get("vector_store") is None and kwargs.get("connection") is None:
+            kwargs["connection"] = PGVector()
+        super().__init__(**kwargs)
+
+    @property
+    def vector_store_cls(self):
+        return PGVectorStore
+
+    @property
+    def vector_store_params(self):
+        return self.model_dump(include=set(BaseWriterVectorStoreParams.model_fields)) | {
+            "connection": self.connection,
+            "client": self.client,
+        }
+
+    def execute(
+        self, input_data: dict[str, Any], config: RunnableConfig = None, **kwargs
+    ):
+        """
+        Execute the document writing operation.
+
+        This method writes the input documents to the PGVector Vector Store.
+
+        Args:
+            input_data (dict[str, Any]): Input data containing the documents to be written.
+            config (RunnableConfig, optional): Configuration for the execution.
+            **kwargs: Additional keyword arguments.
+
+        Returns:
+            dict: A dictionary containing the count of upserted documents.
+        """
+        config = ensure_config(config)
+        self.run_on_node_execute_run(config.callbacks, **kwargs)
+
+        documents = input_data["documents"]
```

Comment:
Please add support of `content_key` and `embedding_key` similar to Milvus (check parent classes too):
https://github.com/dynamiq-ai/dynamiq/blob/c547e44a804fccd94d01954be4060115c97c0b82/dynamiq/nodes/writers/milvus.py#L75

Reply:
Please add `table_name` and `schema_name` as well, as this extra attributes that make sense for each of PGVector node.

Reply:
Added the relevant parameters to base class for `PGVectorStoreParams`.